ui/settings: add setting descriptions

rr- opened the pull request:
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TRX/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

![20250607_225822_Level_0](https://github.com/user-attachments/assets/d78cccc9-592c-4cc4-8882-560fa63a2566)
![20250607_230019_Level_0](https://github.com/user-attachments/assets/5995d832-bafc-4287-8835-3c4d61808eee)
![20250607_225939_Level_0](https://github.com/user-attachments/assets/bb11d563-9436-452f-95b9-1a91a023015c)

- Introduces help to all options except water color.
- Adjusts the game strings linter so that it accepts libtrx using strings, if they are defined in all child projects.
- Renames examine item dialog to text dialog.

#### Testing

This was a semi automated job, so I'd appreciate going into every setting, reading it and letting me know if it makes sense or not. (I'd appreciate suggestions on changes too.)

rr- committed:
ui/settings: add setting descriptions

Richard-L commented:
Please let's not lose a "(PS1)" indicator as it helps to remember and direct people who prefer this. Just like with TombATI which you kept.

rr- commented:
I took some time to understand this comment since it's a bit out of context. However, this pull request focuses on **descriptions**. It's not about removing anything; rather, it showcases how the UI handles lengthy help text. The `1 / 2 →` suggests you can navigate to the second page to see the PS1.

![20250608_010511_Level_0](https://github.com/user-attachments/assets/be2b218d-2bca-4c1f-a251-a79b91fa9552)

aredfan commented on data/tr1/ship/cfg/TR1X_strings.json5:
```diff
@@ -389,92 +389,164 @@
         "GAMEPLAY_SETTINGS_DISABLE_TREX_COLLISION": "Remove dead T-Rex collision",
+        "GAMEPLAY_SETTINGS_DISABLE_TREX_COLLISION_DESCRIPTION": "Removes all collision with T-Rex upon death. This helps when the T-Rex's body blocks the passage out.",
         "GAMEPLAY_SETTINGS_DISABLE_UZIS": "Remove uzis",
+        "GAMEPLAY_SETTINGS_DISABLE_UZIS_DESCRIPTION": "Removes all Uzi guns and Uzi ammo pickups from the game.",
```
Just a quick comment, the `Uzi guns` and `Uzi ammo` could be all lower case to match the other 'remove weapons' descriptions.

rr- replied:
I updated this and other similar options.

aredfan reviewed (commented):
Just a review of TR2 for now.

**Edit:** I somehow messed up my review so I'll write what I noticed below.

- [x] The sound description is mixed up with the `Music volume`.
- [x] The music description is mixed up with the `Sound volume`.
- [x] The `Bar scale` description could perhaps be reworded to `Changes size of health, air and enemy bars.`
- [x] The `Legal screens` option refers to eidos logo and bink video codec fmv, both which isn't present in TR2. It could be reworded to `Enables legal screen and Core Design FMV at the game start.`

**Edit 2:** Please correct me if I'm wrong regarding the last point above. For TR1, the reference to `bink video codec` may be incorrect as this is the format used for TR4+. The FMVs in TR1 uses RPL format and appears to be called `escape`.

<details><summary>Click here to see the escape FMV screenshot</summary>

![1](https://github.com/user-attachments/assets/7b8eadc6-90a2-4388-8082-2c063cd0c9d8)

</details>

lahm86 commented on src/libtrx/game/ui/dialogs/sound_settings.c:
```diff
@@ -27,6 +27,7 @@ static const UI_SETTINGS_OPTION m_SoundOptions[] = {
     {
         .option_type = COT_FLOAT,
         .label_id = GS_ID(SOUND_SETTINGS_SOUND_VOLUME),
+        .description_id = GS_ID(SOUND_SETTINGS_SOUND_VOLUME_DESCRIPTION),
```
The description here is swapped with the music volume description.

lahm86 replied:
aredfan caught it, apologies for the dupe.

lahm86 commented on src/libtrx/game/ui/dialogs/sound_settings.c:
```diff
@@ -41,6 +42,7 @@ static const UI_SETTINGS_OPTION m_SoundOptions[] = {
     {
         .option_type = COT_FLOAT,
         .label_id = GS_ID(SOUND_SETTINGS_MUSIC_VOLUME),
+        .description_id = GS_ID(SOUND_SETTINGS_MUSIC_VOLUME_DESCRIPTION),
```
Swapped with above.

lahm86 reviewed (commented):
Looks great! Just spotted one text issue below.

A thought - if it's not too much bother to implement - could we respond to `menu_back` input while the description window is open to toggle it off? Just out of habit I was doing this and it bubbles up to close the settings dialog itself. Minor thing really.

Richard-L commented:
> I took some time to understand this comment since it's a bit out of context. However, this pull request focuses on **descriptions**. It's not about removing anything; rather, it showcases how the UI handles lengthy help text. The `1 / 2 →` suggests you can navigate to the second page to see the PS1.
> 
> ![20250608_010511_Level_0](https://private-user-images.githubusercontent.com/1045476/452658242-be2b218d-2bca-4c1f-a251-a79b91fa9552.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkzNzc3ODcsIm5iZiI6MTc0OTM3NzQ4NywicGF0aCI6Ii8xMDQ1NDc2LzQ1MjY1ODI0Mi1iZTJiMjE4ZC0yYmNhLTRjMWYtYTI1MS1hNzliOTFmYTk1NTIuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDhUMTAxMTI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjkxZjM2Y2FmOTY4YTc1NDc0ODA1Y2M2ZDhmMjJmNTc0NWQ2ZTgxZjQ1NmVkYjY3ODk4ZjBjNzY1ZDI0ODk0OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ZO9o69btjaYf4FJTAw2Ig0_902CUxOFIK6Bt5XGkGo4)

I see you put this option on the second page. I missed that. But I'm sad you chose to put it last, even after "challenge run" stuff. Please consider putting it right under what you call the "TombATI" setting. The PS1 version is how the majority of people originally experienced the first two games when they were first released. The visibility of the options should reflect that.

More importantly: the health bar behaviour described here as PS1 only applies to TR1, so it's wrong here. On PS1 TR2 the health bar flashes and always shows when <= 20%.

Some general comments:
- I applaud the effort of wanting to bring everything in-game and lose the config tool. But you're using and forging a UI that was never intended for reading text, and so there are immediate shortcomings:
  - No indentation after bullet points
  - Lack of heading and sub-heading clarity (you tried to circumvent this with an UPPERCASE)
  - Lack of margin after paragraphs
  - Lack of a natural arrangement of windows (currently they sit on top making the black even blacker (?), and with some of the underneath still showing? "toggle help")
  - Lack of indication how to go back to the parent menu (where even without this level of complexity the console releases had this feature which now seems more urgent than ever #545 ).
- "TombATI" has nothing to do with TR2, so it's surprising to see it here. A new term would be better imo, perhaps "TRX default". Generally, I understand you want to move away from a per-game understanding and merge everything under one TRX umbrella. I find that regrettable for several reasons. It inevitably mixes facts to a point where they're no longer discernable. It already happened here where a TR1-only behaviour is labelled as a PS1 option for TR2. The faux-Japanese game modes are another example. I'm really sad this kind of respect for the OGs is lost more and more. Personally I think it's important to get these facts correct and only show them where they truthfully apply.
- Can the seperation indication of behaviour be consistent: Either put the version at the beginning (like you did with PS1), or put it at the end in brackets (like you did with TombATI).

I think the text generally needs rewording and simplification, ideally eliminating the need for a second page (at this UI scale).

rr- commented:
> Just a review of TR2 for now.
> 
> **Edit:** I somehow messed up my review so I'll write what I noticed below.
> 
> * [ ]  The sound description is mixed up with the `Music volume`.
> * [ ]  The music description is mixed up with the `Sound volume`.
> * [ ]  The `Bar scale` description could perhaps be reworded to `Changes size of health, air and enemy bars.`
> * [ ]  The `Legal screens` option refers to eidos logo and bink video codec fmv, both which isn't present in TR2. It could be reworded to `Enables legal screen and Core Design FMV at the game start.`
> 
> **Edit 2:** Please correct me if I'm wrong regarding the last point above. For TR1, the reference to `bink video codec` may be incorrect as this is the format used for TR4+. The FMVs in TR1 uses RPL format and appears to be called `escape`.

Good catch! There was a mistake in there, TR1 uses RPL format. Changed the wording for both games to be more agnostic as per the original suggestion.

Fixed the rest of the issues, including the `menu_back` issue (Thanks Lahm!)

rr- commented:
> > I took some time to understand this comment since it's a bit out of context. However, this pull request focuses on **descriptions**. It's not about removing anything; rather, it showcases how the UI handles lengthy help text. The `1 / 2 →` suggests you can navigate to the second page to see the PS1.
> > ![20250608_010511_Level_0](https://private-user-images.githubusercontent.com/1045476/452658242-be2b218d-2bca-4c1f-a251-a79b91fa9552.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkzNzc3ODcsIm5iZiI6MTc0OTM3NzQ4NywicGF0aCI6Ii8xMDQ1NDc2LzQ1MjY1ODI0Mi1iZTJiMjE4ZC0yYmNhLTRjMWYtYTI1MS1hNzliOTFmYTk1NTIuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDhUMTAxMTI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjkxZjM2Y2FmOTY4YTc1NDc0ODA1Y2M2ZDhmMjJmNTc0NWQ2ZTgxZjQ1NmVkYjY3ODk4ZjBjNzY1ZDI0ODk0OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ZO9o69btjaYf4FJTAw2Ig0_902CUxOFIK6Bt5XGkGo4)
> 
> I see you put this option on the second page. I missed that. But I'm sad you chose to put it last, even after "challenge run" stuff. Please consider putting it right under what you call the "TombATI" setting. The PS1 version is how the majority of people originally experienced the first two games when they were first released. The visibility of the options should reflect that.
> 
> More importantly: the health bar behaviour described here as PS1 only applies to TR1, so it's wrong here. On PS1 TR2 the health bar flashes and always shows when <= 20%.
> 
> Some general comments:
> 
> * I applaud the effort of wanting to bring everything in-game and lose the config tool. But you're using and forging a UI that was never intended for reading text, and so there are immediate shortcomings:
>   
>   * No indentation after bullet points
>   * Lack of heading and sub-heading clarity (you tried to circumvent this with an UPPERCASE)
>   * Lack of margin after paragraphs
>   * Lack of a natural arrangement of windows (currently they sit on top making the black even blacker (?), and with some of the underneath still showing? "toggle help")
>   * Lack of indication how to go back to the parent menu (where even without this level of complexity the console releases had this feature which now seems more urgent than ever [Feature: PS1 style "Select" and "Go Back" indicators #545](https://github.com/LostArtefacts/TRX/issues/545) ).
> * "TombATI" has nothing to do with TR2, so it's surprising to see it here. A new term would be better imo, perhaps "TRX default". Generally, I understand you want to move away from a per-game understanding and merge everything under one TRX umbrella. I find that regrettable for several reasons. It inevitably mixes facts to a point where they're no longer discernable. It already happened here where a TR1-only behaviour is labelled as a PS1 option for TR2. The faux-Japanese game modes are another example. I'm really sad this kind of respect for the OGs is lost more and more. Personally I think it's important to get these facts correct and only show them where they truthfully apply.
> * Can the seperation indication of behaviour be consistent: Either put the version at the beginning (like you did with PS1), or put it at the end in brackets (like you did with TombATI).
> 
> I think the text generally needs rewording and simplification, ideally eliminating the need for a second page (at this UI scale).

I’ll consider some of the points you’ve raised. Just a heads up—when feedback leans heavily into criticism without offering concrete, actionable suggestions, it’s hard to do anything useful with it.

I’m sorry to hear that you feel disappointed, but at the end of the day, that’s not something I can or should take responsibility for. We’re going to continue developing according to our roadmap and values. Thanks.

lahm86 commented on src/libtrx/include/libtrx/game/game_string.def:
```diff
@@ -398,3 +398,128 @@ GS_DEFINE(GRAPHIC_SETTINGS_ENABLE_ROUND_SHADOW,        "Round shadows")
+GS_DEFINE(GAMEPLAY_SETTINGS_FIX_WALL_JUMP_GLITCH_DESCRIPTION,           "Fixes Lara jumping through objects and moving up certain slopes.")
+GS_DEFINE(GAMEPLAY_SETTINGS_LOAD_MUSIC_TRIGGERS_DESCRIPTION,            "Loads previously triggered, one shot music so one shot music tracks do not replay.")
+GS_DEFINE(GAMEPLAY_SETTINGS_MAXIMUM_SAVE_SLOTS_DESCRIPTION,             "Changes the number of available save slots.")
+GS_DEFINE(GAMEPLAY_SETTINGS_REMEMBER_GUNS_BETWEEN_LEVELS_DESCRIPTION,   "Makes Lara revert to pistols at the start of each level.")
```
Perhaps wording here would be better as follows now that it's inverted (I forgot about this earlier, apologies).
`Makes Lara remember which gun she was using last in the previous level when starting a new level. If disabled, Lara will revert to pistols.`

rr- replied:
Thanks, updated

lahm86 commented on src/tr1/game/game_string.def:
```diff
@@ -18,3 +18,11 @@ GS_DEFINE(MISC_EMPTY_SLOT_FMT, "- EMPTY SLOT %d -")
+GS_DEFINE(GAMEPLAY_SETTINGS_ENABLE_GAME_MODES_DESCRIPTION,              "Allows new game plus options to be selected from the new game passport menu.\n\n- New Game+: unlocks all weapons with infinite ammo; enemies have double the HP.\n- Japanese NG: weapons do double damage.\n- Japanese NG+: combination of New Game+ and Japanese NG.")
+GS_DEFINE(GRAPHIC_SETTINGS_ENABLE_GUN_LIGHTING_DESCRIPTION,             "Enables dynamic lighting to be generated for gunshots and explosions, similar to TR2+.")
+GS_DEFINE(GRAPHIC_SETTINGS_ENABLE_FADE_EFFECTS_DESCRIPTION,             "Makes the inventory and pause screen use fade transitions for the semitransparent backdrop, similar to PS1.")
+GS_DEFINE(GAMEPLAY_SETTINGS_FIX_FLOOR_DATA_ISSUES_DESCRIPTION,          "Fixes original issues with floor data/triggers, including the following.\n\n- The untriggered bat in Colosseum.\n- The secret in Great Pyramid that does not register.")
```
We should perhaps drop the bullet list as it's now so extensive and covered in detail in the README.

rr- replied:
Thanks, updated

rr- committed:
ui/text: indent bullet lists

lahm86 reviewed (commented):
Just two additional text suggestions, otherwise this looks awesome.

aredfan commented:
The previous issues are all resolved.  Thank you. 👍 

Just a few new issues:

- [x] Since the `Healthbar mode` and `Airbar mode` descriptions have been partially rewritten, it'll be worth ensuring the toggable options matches the description's order e.g. the `Never` option comes last.
- [x] The `Healthbar mode` has `Default or flashing` as an option, conversely it displays `Flashing or default` in the description.

rr- committed:
ui/text: tidy

rr- commented:
> The previous issues are all resolved. Thank you. 👍
> 
> Just a few new issues:
> 
> * [ ]  Since the `Healthbar mode` and `Airbar mode` descriptions have been partially rewritten, it'll be worth ensuring the toggable options matches the description's order e.g. the `Never` option comes last.
> * [ ]  The `Healthbar mode` has `Default or flashing` as an option, conversely it displays `Flashing or default` in the description.


Thanks – updated

aredfan commented:
One last review 😌 

- [x] TR1: In Gameplay/Fixes, the `Fix bridge collision` option appears twice. (4th and 7th line)
- [x] TR1: In Graphics/UI, the `Smooth bars` option description could be reworded to `Make the health, air and enemy bars use smooth color transitions.`
- [x] TR1: In Gameplay/Mods, the `Persistent damage` option description could be reworded to `Stops Lara from healing when starting a new level (for no Heal challenge runs).`
- [x] TR1: In Gameplay/General, the `Save crystals` option description's first sentence could be reworded to `Limits saving to the beginning of levels and save crystals.`  Note the extra `and` in this sentence.

rr- commented:
> One last review 😌
> 
> * [ ]  TR1: In Gameplay/Fixes, the `Fix bridge collision` option appears twice. (4th and 7th line)
> * [ ]  TR1: In Graphics/UI, the `Smooth bars` option description could be reworded to `Make the health, air and enemy bars use smooth color transitions.`
> * [ ]  TR1: In Gameplay/Mods, the `Persistent damage` option description could be reworded to `Stops Lara from healing when starting a new level (for no Heal challenge runs).`
> * [ ]  TR1: In Gameplay/General, the `Save crystals` option description's first sentence could be reworded to `Limits saving to the beginning of levels and save crystals.`  Note the extra `and` in this sentence.

Thanks – fixed all points.

aredfan reviewed (approved):
> Thanks – fixed all points.

Thank you. All LGTM. 👍

Btw I noticed the `Fix alligator AI` option is missing from Gameplay/Fixes in TR1.

rr- commented:
> > Thanks – fixed all points.
> 
> Thank you. All LGTM. 👍
> 
> Btw I noticed the `Fix alligator AI` option is missing from Gameplay/Fixes in TR1.

Excellent catch. It was a silly typo (`TR__VERSION == 1` instead of `TR_VERSION == 1`). Fixed.
